fingerprints/fingerprints.yaml

```diff
@@ -0,0 +1,821 @@
+      - pattern: "#!/usr/bin/env ... bash"
+    severity: INVENTORY
+  - id: cai-shell-technology-sh
+    languages:
+      - generic
+    message: Detected a sh script. This likely indicates sh is being used.
+    metadata:
+      license: Semgrep Registry License
+    pattern-either:
+      - pattern: "#!/bin/sh"
+      - pattern: "#!/usr/bin/env ... sh"
+    severity: INVENTORY
+  - id: cai-c-technology-c
+    languages:
+      - generic
+    message: Detected a .c file. This likely indicates C is in use.
+    metadata:
+      license: Semgrep Registry License
+    paths:
+      include:
+        - "*.c"
+    pattern-regex: ^.
+    severity: INVENTORY
+  - id: cai-csharp-technology-csharp
+    languages:
+      - generic
+    message: Detected a .cs file. This likely indicates C is in use.
+    metadata:
+      license: Semgrep Registry License
+    paths:
+      include:
+        - "*.cs"
+    pattern-regex: ^.
+    severity: INVENTORY
+  - id: cai-go-technology-go
+    languages:
+      - generic
+    message: Detected a .go file. This likely indicates Go is in use.
+    metadata:
+      license: Semgrep Registry License
+    paths:
+      include:
+        - "*.go"
+    pattern-regex: ^.
+    severity: INVENTORY
+  - id: cai-html-technology-html
+    languages:
+      - generic
+    message: Detected a .html file. This likely indicates html is in use.
+    metadata:
+      license: Semgrep Registry License
+    paths:
+      include:
+        - "*.html"
+    pattern-regex: ^.
+    severity: INVENTORY
+  - id: cai-java-technology-java
+    languages:
+      - generic
+    message: Detected a .java file. This likely indicates Java is in use.
+    metadata:
+      license: Semgrep Registry License
+    paths:
+      include:
+        - "*.java"
+    pattern-regex: ^.
+    severity: INVENTORY
+  - id: cai-javascript-technology-javascript
+    languages:
+      - generic
+    message: Detected a .js file. This likely indicates JavaScript is in use.
+    metadata:
+      license: Semgrep Registry License
+    paths:
+      include:
+        - "*.js"
+        - "*.javascript"
+    pattern-regex: ^.
+    severity: INVENTORY
+  - id: cai-kotlin-technology-kotlin
+    languages:
+      - generic
+    message: Detected a .kt file. This likely indicates Kotlin is in use.
+    metadata:
+      license: Semgrep Registry License
+    paths:
+      include:
+        - "*.kt"
+    pattern-regex: ^.
+    severity: INVENTORY
+  - id: cai-ocaml-technology-ocaml
+    languages:
+      - generic
+    message: Detected a .ml file. This likely indicates Ocaml is in use.
+    metadata:
+      license: Semgrep Registry License
+    paths:
+      include:
+        - "*.ml"
+    pattern-regex: ^.
+    severity: INVENTORY
+  - id: cai-php-technology-php
+    languages:
+      - generic
+    message: Detected a .php file. This likely indicates PHP is in use.
+    metadata:
+      license: Semgrep Registry License
+    paths:
+      include:
+        - "*.php"
```

Comment:
I think PHP files also end in php4 or php5 or something? Looked it up: https://www.guru99.com/what-is-php-first-php-program.html

.php3
.php4
.php5
.phps

Could also look for `<?php ...>`